pkg/tasks/task.go

```diff
@@ -45,7 +45,7 @@ type Task struct {
 // Run runs a task
 func (t *Task) Run(s *state.State) error {
 	if t.Retries == 0 {
-		t.Retries = 3
+		t.Retries = 10
```

Comment:
I don't think that it makes a lot of sense to increase the number of retries to 10 by default. That seems way too high for many tasks. I think we should do that only for specific tasks.

Also, the problem mentioned in the linked issue is supposed to be mitigated with #994. Do we have some idea did that fix helped?

Reply:
I think 10 is too small actually, checkout the number of retries in kubermatic code (especially shell) for example. Yes, maybe some parts don't have too much sense to retry ever 2 times. I prefer to have many retries by default and maybe exclude some parts to opposite approach (no or small retries, with some parts explicitly have big retries).

Reply:
>Do we have some idea did that fix helped?

It was initially known that elevated number of retries would fix the problem.

Reply:
> I think 10 is too small actually, checkout the number of retries in kubermatic code (especially shell) for example.

The number of retries depend on how the functionality is implemented. Note that we use the exponential backoff, so with each retry, the user is going to wait a bit more each time.

> Yes, maybe some parts don't have too much sense to retry ever 2 times. I prefer to have many retries by default and maybe exclude some parts to opposite approach (no or small retries, with some parts explicitly have big retries).

That's why I think it's best to do this on the task level and keep some sane default number of retries. I can think of a few tasks that don't make sense to retry a lot, like installing the binaries or running kubeadm.

On top of that, I think it's important to understand how each task works and to make them behave consistently. For example, some tasks already have implemented retries in the task logic, so adding more retries would just increase the time spent on useless waiting.

Reply:
>and keep some sane default number of retries

Totally agree. I just see that the sane number is bigger then 3, because 3 retries are "spent" superfast even with exponential backoff, simply because there is no room for exponent to be considerable long.

Reply:
>installing the binaries
>running kubeadm

Both case fall because of intermediate networking issues, like for example temporary down DNS that will be up in 30 seconds. This is exact demonstration why we should keep this number relatively high by default. Because "we don't know" in what unpredictable way everything may go south "in this particular instance".